compiler/test/suites/imports.re

```diff
@@ -15,175 +15,121 @@ describe("imports", ({test, testSkip}) => {
+    "import_alias_multiple",
+    "include \"provideStar\" as ProvideStar; from ProvideStar use {x as y, y as x}; x(y)",
   );
   /* import {} errors */
   assertCompileError(
     "import_some_error",
-    "import {a} from \"exportStar\"; a",
-    "Export \"a\" was not found in \"exportStar\"",
+    "include \"provideStar\" as ProvideStar; from ProvideStar use {a}; a",
+    "Unbound value a in module ProvideStar",
   );
   assertCompileError(
     "import_some_error2",
-    "import {x, a} from \"exportStar\"; a",
-    "Export \"a\" was not found in \"exportStar\"",
+    "include \"provideStar\" as ProvideStar; from ProvideStar use {x, a}; a",
+    "Unbound value a in module ProvideStar",
   );
   assertCompileError(
     "import_some_error3",
-    "import {Foo} from \"exportStar\"; a",
-    "Export \"Foo\" was not found in \"exportStar\"",
+    "include \"provideStar\" as ProvideStar; from ProvideStar use {Foo}; a",
+    "Unbound module Foo in module ProvideStar",
   );
   assertCompileError(
     "import_some_error3",
-    "import {x, Foo} from \"exportStar\"; a",
-    "Export \"Foo\" was not found in \"exportStar\"",
+    "include \"provideStar\" as ProvideStar; from ProvideStar use {x, Foo}; a",
+    "Unbound module Foo in module ProvideStar",
   );
   /* import module tests */
-  assertSnapshot("import_module", "import Foo from \"exportStar\"; Foo.x");
+  assertSnapshot("import_module", "include \"provideStar\" as Foo; Foo.x");
   assertSnapshot(
     "import_module2",
-    "import Foo from \"exportStar\"; Foo.y(Foo.x)",
+    "include \"provideStar\" as Foo; Foo.y(Foo.x)",
   );
   /* import module errors */
   assertCompileError(
     "import_module_error",
-    "import Foo from \"exportStar\"; Foo.foo",
+    "include \"provideStar\" as Foo; Foo.foo",
     "Unbound value foo in module Foo",
   );
   /* import well-formedness errors */
   assertCompileError(
     "import_alias_illegal_renaming",
-    "import {Cons as cons, Empty} from \"list\"; cons(3, Empty)",
-    "Alias 'cons' should have proper casing",
+    "include \"list\" as List; from List use {Cons as cons, Empty}; cons(3, Empty)",
+    "Expected an uppercase module alias",
   );
   assertCompileError(
     "import_alias_illegal_renaming2",
-    "import {sum as Sum, Empty} from \"list\"; sum(Empty)",
-    "Alias 'Sum' should have proper casing",
+    "include \"list\" as List; from List use {sum as Sum, Empty}; sum(Empty)",
+    "Expected a lowercase alias",
   );
   assertCompileError(
     "import_module_illegal_name",
-    "import foo from \"list\";",
-    "Module 'foo' should have an uppercase name",
+    "include \"list\" as foo;",
+    "Expected an uppercase module identifier",
   );
   assertCompileError(
     "import_module_not_external",
-    "import Foo.Foo from \"list\";",
-    "Module name 'Foo.Foo' should contain only one module.",
+    "include \"list\" as Foo.Foo;",
+    "Expected a newline character to terminate the statement",
```

Comment:
This is an interesting error for the problem. Anyway we can make it better?

Reply:
We don't test the exact location here in the test, but the error shows up on the `.` in `Foo.Foo` and says that it was unexpected, which I don't think is too bad of an error.

Before this PR we just parsed the whole identifier here, but the bad thing about that is we end up with a bunch of `Impossible` throughout the compiler. We could maybe go back to parsing it the old way (though slightly better now) and in the action raise a SyntaxError if the user did this and extract the single module name.

Reply:
Hmm, I'm on the fence about having Impossible cases throughout the compiler but better error messages are always appreciated 🤔